functional-tests/fn_bridge_withdraw_happy.py

```diff
@@ -0,0 +1,184 @@
+
+        self.logger.debug(f"BTC URL: {btc_url}")
+        self.logger.debug(f"BTC user: {btc_user}")
+        self.logger.debug(f"BTC password: {btc_password}")
+
+        web3: Web3 = reth.create_web3()
+        web3.eth.default_account = web3.address
+        el_address = web3.address
+        self.logger.debug(f"EL address: {el_address}")
+
+        # Get operators pubkey and musig2 aggregates it
+        bridge_pk = get_bridge_pubkey(seqrpc)
+        self.logger.debug(f"Bridge pubkey: {bridge_pk}")
+
+        seq_addr = seq.get_prop("address")
+        self.logger.debug(f"Sequencer Address: {seq_addr}")
+        bridge_pk = get_bridge_pubkey(seqrpc)
+
+        # Generate Plenty of BTC to address
+        btcrpc.proxy.generatetoaddress(102, address)
+
+        # Deposit to the EL address
+        # NOTE: we need 2 deposits to make sure we have funds for gas
+        self.make_drt(ctx, el_address, bridge_pk)
+        self.make_drt(ctx, el_address, bridge_pk)
+        time.sleep(0.5)
+
+        # Send funds to the bridge precompile address for a withdrawal
+        change_address_pk = extract_p2tr_pubkey(change_address)
+        self.logger.debug(f"Change Address PK: {change_address_pk}")
+        l2_tx_hash = self.make_withdraw(ctx, change_address_pk, gas).hex()
+        self.logger.debug(f"Sent withdrawal transaction with hash: {l2_tx_hash}")
+
+        # Wait for the withdrawal to be processed
+        time.sleep(DEFAULT_BLOCK_TIME_SEC * 2)
+
+        l2_tx_state = web3.eth.get_transaction_receipt(l2_tx_hash)
+        self.logger.debug(f"Transaction state (web3): {l2_tx_state}")
+
+        # Make sure that the balance is zero
+        balance = int(rethrpc.eth_getBalance(el_address), 16)
+        self.logger.debug(f"Balance after withdrawal (EL address): {balance}")
+
+        # FIXME: I don't know why the hell the math is wrong
+        #        and I need to manually add 5 zeros with 100_000
+        #
+        # lower bound is 10 BTC - 0.01 BTC (fees) - gas with a leeway
+        lower_bound = ((rollup_deposit_amount - 1_000_000) * SATS_TO_WEI * 100_000) - gas * 2
+        # upper bound is 10 BTC + 0.01 BTC (fees)
+        upper_bound = (rollup_deposit_amount + 1_000_000) * SATS_TO_WEI * 100_000
+        self.logger.debug(f"Balance lower bound: {lower_bound}")
+        self.logger.debug(f"Balance upper bound: {upper_bound}")
+
+        assert (
+            balance >= lower_bound and balance <= upper_bound
+        ), "balance is not in the expected range"
+
+        # Mine blocks
+        btcrpc.proxy.generatetoaddress(12, UNSPENDABLE_ADDRESS)
+        time.sleep(0.5)
+
+        # Make sure that the balance in the BTC wallet is 10 BTC-ish
+        btc_balance = get_balance(change_address, btc_url, btc_user, btc_password)
+        self.logger.debug(f"BTC balance: {btc_balance}")
+        # FIXME: This only gives me 4.97 BTC WTF?
+        assert btc_balance >= 9_900_000_000, "BTC balance is not 10 BTC"
```

Comment:
We should get `10 BTC`-ish back.
But how the hell I got less than half?
Cc @delbonis